web/crux/src/services/email-builder.service.ts

```diff
@@ -0,0 +1,51 @@
+import { Injectable } from '@nestjs/common'
+import { disassembleKratosRecoveryUrl } from 'src/domain/utils'
+import { EmailItem } from './email.service'
+
+type InviteTemaple = {
+  subject: string
+  text: string
+  html: string
+}
+
+const HOST = process.env.CRUX_UI_URL
+const sender = { email: process.env.FROM_EMAIL, name: process.env.FROM_NAME }
+
+@Injectable()
+export class EmailBuilder {
+  buildInviteEmail(email: string, teamName: string, teamId?: string, kratosRecoveryLink?: string): EmailItem | null {
+    if (!teamId && !kratosRecoveryLink) {
+      return null
+    }
+
+    const inviteTemplate = this.getInviteTemplate(teamName, teamId, kratosRecoveryLink)
+
+    const emailItem: EmailItem = {
+      sender: sender,
+      recipient: email,
+      subject: inviteTemplate.subject,
+      text: inviteTemplate.text,
+      html: inviteTemplate.html,
+    }
+
+    return emailItem
+  }
+
+  private getInviteTemplate(teamName: string, teamId?: string, kratosRecoveryLink?: string): InviteTemaple {
+    const link = teamId ? `${HOST}/teams/${teamId}/invite` : disassembleKratosRecoveryUrl(HOST, kratosRecoveryLink)
+    const mode = teamId ? 'to accept' : 'to accept and create a dyrector.io account,'
+    const button = teamId ? 'Accept' : 'Create account'
```

Comment:
It would be better if we check `teamId` only once and create a full object based on the output.